gdx/src/com/badlogic/gdx/utils/ObjectMap.java

```diff
@@ -1,846 +1,779 @@
+//		for (Entry<K, V> entry : map)
+//			put(entry.key, entry.value);
+//	}
+	public void putAll (ObjectMap<K, V> map) {
+		ensureCapacity(map.size);
+		final K[] keyTable = map.keyTable;
+		final V[] valueTable = map.valueTable;
+		K k;
+		for (int i = 0, n = keyTable.length; i < n; i++) {
+			if ((k = keyTable[i]) != null)
+				put(k, valueTable[i]);
+		}
+	}
+
+	/**
+	 * Skips checks for existing keys.
+	 */
+	private void putResize (K key, V value) {
+		K[] keyTable = this.keyTable;
+		V[] valueTable = this.valueTable;
+		int b = place(key);
+		for (int i = b; ; i = (i + 1) & mask) {
+			// space is available so we insert and break (resize is later)
+			if (keyTable[i] == null) {
+				keyTable[i] = key;
+				valueTable[i] = value;
+				break;
+			}
+		}
+		if (++size >= threshold) {
+			resize(keyTable.length << 1);
+		}
+	}
+
+	/**
+	 * Returns the value for the specified key, or null if the key is not in the map.
+	 */
+	public V get (K key) {
+		final int loc = locateKey(key);
+		return loc == -1 ? null : valueTable[loc];
+	}
+
+	/**
+	 * Returns the value for the specified key, or the default value if the key is not in the map.
+	 */
+	public V get (K key, V defaultValue) {
+		final int loc = locateKey(key);
+		return loc == -1 ? defaultValue : valueTable[loc];
+	}
+
+	public V remove (K key) {
+		int loc = locateKey(key);
+		if (loc == -1) {
+			return null;
+		}
+		V oldValue = valueTable[loc];
+		while ((key = keyTable[loc + 1 & mask]) != null && (loc + 1 & mask) != place(key)) {
+			keyTable[loc] = key;
+			valueTable[loc] = valueTable[++loc & mask];
+		}
+		keyTable[loc] = null;
+		valueTable[loc] = null;
+		--size;
+		return oldValue;
+	}
+
+	/**
+	 * Returns true if the map has one or more items.
+	 */
+	public boolean notEmpty () {
+		return size > 0;
+	}
+
+	/**
+	 * Returns true if the map is empty.
+	 */
+	public boolean isEmpty () {
+		return size == 0;
+	}
+
+	/**
+	 * Reduces the size of the backing arrays to be the specified capacity or less. If the capacity is already less, nothing is
+	 * done. If the map contains more items than the specified capacity, the next highest power of two capacity is used instead.
+	 */
+	public void shrink (int maximumCapacity) {
+		if (maximumCapacity < 0)
+			throw new IllegalArgumentException("maximumCapacity must be >= 0: " + maximumCapacity);
+		if (size > maximumCapacity)
+			maximumCapacity = size;
+		if (keyTable.length <= maximumCapacity)
+			return;
+		resize(MathUtils.nextPowerOfTwo(maximumCapacity));
+	}
+
+	/**
+	 * Clears the map and reduces the size of the backing arrays to be the specified capacity if they are larger.
+	 */
+	public void clear (int maximumCapacity) {
+		if (keyTable.length <= maximumCapacity) {
+			clear();
+			return;
+		}
+		size = 0;
+		resize(maximumCapacity);
+	}
+
+	public void clear () {
+		if (size == 0)
+			return;
+		K[] keyTable = this.keyTable;
+		V[] valueTable = this.valueTable;
+		for (int i = keyTable.length; i > 0; ) {
+			keyTable[--i] = null;
```

Comment:
for (int i = keyTable.length; i > 0; --i) instead of decreasing in loop, not needed here to decrease in loop only or use a while instead bute here => for (int i = keyTable.length; i > 0; --i) or for (int i = keyTable.length; i-- > 0; ) should be faster